packages/input-selection/src/ChangeAddress/StaticChangeAddressResolver.ts

```diff
@@ -0,0 +1,40 @@
+import { Cardano } from '@cardano-sdk/core';
+import { ChangeAddressResolver, Selection } from '../';
+import { GroupedAddress } from '@cardano-sdk/key-management';
+import { InvalidStateError } from '@cardano-sdk/util';
+import { Observable, firstValueFrom } from 'rxjs';
+
+/**
+ * Default change address resolver.
+ */
+export class StaticChangeAddressResolver implements ChangeAddressResolver {
+  readonly #addresses$: Observable<GroupedAddress[]>;
+
+  /**
+   * Initializes a new instance of the StaticChangeAddressResolver.
+   *
+   * @param addresses$ The observable with the known addresses.
+   */
+  constructor(addresses$: Observable<GroupedAddress[]>) {
+    this.#addresses$ = addresses$;
+  }
+
+  /**
+   * Always resolves to the same address.
+   */
+  async resolve(selection: Selection): Promise<Array<Cardano.TxOut>> {
+    const updatedSelection = { ...selection };
```

Comment:
I don't think that creating the `updatedSelection` is needed. You only need to calculate the change based on the input `selection.change`:
```diff
-    const updatedSelection = { ...selection };
    const groupedAddresses = await firstValueFrom(this.#addresses$);
....
+ return updatedSelection.change.map((txOut) => ({...txout, address}));
```